sw/airborne/mcu.c

```diff
@@ -89,11 +89,19 @@ void mcu_init(void)
 #if defined MCU_PWR
   gpio_setup_output(MCU_PWR, MCU_PWR_PIN);
 #if defined BTN_ON
-  if(!gpio_get(BTN_ON, BTN_ON_PIN))
+  gpio_setup_input(BTN_ON, BTN_ON_PIN);
+  if(gpio_get(BTN_ON, BTN_ON_PIN))
 #endif
   {
-  MCU_PWR_ON(MCU_PWR, MCU_PWR_PIN);
+    MCU_PWR_ON(MCU_PWR, MCU_PWR_PIN);
   }
+#if defined BTN_ON
+  else {
+    while(true) {
```

Comment:
Else other parts of the software can still turn it on later. This is just to make sure that the on button needs to be pressed in order to turn it on.
Because of our schematic we were able to turn the autopilot on by moving the servo's previously.